lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/driver/AzureBatchTaskStatusAlarmHandler.java

```diff
@@ -0,0 +1,114 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runime.azbatch.driver;
+
+import com.microsoft.azure.batch.protocol.models.CloudTask;
+import org.apache.reef.runime.azbatch.util.AzureBatchHelper;
+import org.apache.reef.runime.azbatch.util.TaskStatusMapper;
+import org.apache.reef.runtime.common.driver.resourcemanager.ResourceStatusEvent;
+import org.apache.reef.runtime.common.driver.resourcemanager.ResourceStatusEventImpl;
+import org.apache.reef.tang.InjectionFuture;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.time.Clock;
+import org.apache.reef.wake.time.event.Alarm;
+import org.apache.reef.runtime.common.driver.evaluator.pojos.State;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/*
+ * Class that gets that status of the tasks from Azure Batch for the job that is currently in progress
+ * and notifies REEF of the status.
+ */
+final class AzureBatchTaskStatusAlarmHandler implements EventHandler<Alarm> {
+
+  private final AzureBatchHelper azureBatchHelper;
+  private final InjectionFuture<AzureBatchResourceManagerStartHandler> azureBatchResourceManagerStartHandler;
+  private final InjectionFuture<AzureBatchResourceManager> azureBatchResourceManager;
+  private final InjectionFuture<REEFEventHandlers> reefEventHandlers;
+  private final int taskStatusCheckPeriod;
+  private boolean isAlarmScheduled;
+  private Clock clock;
+
+  private static final Logger LOG = Logger.getLogger(AzureBatchResourceLaunchHandler.class.getName());
+
+  @Inject
+  private AzureBatchTaskStatusAlarmHandler(
+      final InjectionFuture<REEFEventHandlers> reefEventHandlers,
+      final AzureBatchHelper azureBatchHelper,
+      final InjectionFuture<AzureBatchResourceManager> azureBatchResourceManager,
+      final InjectionFuture<AzureBatchResourceManagerStartHandler> azureBatchResourceManagerStartHandler,
+      @Parameter(AzureBatchTaskStatusCheckPeriod.class) final int taskStatusCheckPeriod,
+      final Clock clock) {
+    this.reefEventHandlers = reefEventHandlers;
+    this.azureBatchHelper = azureBatchHelper;
+    this.azureBatchResourceManager = azureBatchResourceManager;
+    this.azureBatchResourceManagerStartHandler = azureBatchResourceManagerStartHandler;
+    this.clock = clock;
+    this.taskStatusCheckPeriod = taskStatusCheckPeriod;
+  }
+
+  @Override
+  public void onNext(final Alarm alarm) {
+    String jobId = this.azureBatchHelper.getAzureBatchJobId();
```

Comment:
I don't believe this changes through the lifetime of the object. Suggest moving to the constructor.